Samples/excel-data-types-explorer/src/taskpane/taskpane.ts

```diff
@@ -0,0 +1,1023 @@
+/** Copyright (c) Microsoft Corporation. Licensed under the MIT License. */
+
+// Set up the task pane buttons and select list.
+Office.onReady((info) => {
+  if (info.host === Office.HostType.Excel) {
+    // Assign the HTML buttons to the relevant method.
+    document.getElementById("getData").onclick = getData;
+    document.getElementById("setData").onclick = setData;
+    document.getElementById("clearForm").onclick = clearForm;
+    console.log("test test");
+
+    // Update the displayed input fields based on the select list. 
+    $("#dataTypeSelect").on("change", function() {
+      var value = $("#dataTypeSelect option:selected");
+      setSelectedType(getTypeContent(value.val().toString()));
+    });
+
+    setSelectedType(getTypeContent(defaultType));
+  }
+});
+
+const defaultType: string = "String";
+
+function textInputWithLabel(inputID: string, labelText: string, altText: string): JQuery<HTMLElement>[] {
+  return [
+    $("<td>").append($("<label/>", { for: inputID, class: "labels" }).text(labelText)),
+    $("<td>").append($("<input/>", { class: "inputBox " + inputID, type: "text", id: inputID, alt: altText }))
+  ];
+}
+
+function checkboxWithLabel(inputID: string, labelText: string, altText: string): JQuery<HTMLElement> {
+  return $("<div/>", { class: "checkboxes" })
+    .append($("<input/>", { type: "checkbox", class: inputID, checked: true, alt: altText }))
+    .append($("<label/>").text(labelText));
+}
+
+function stringContent(): JQuery<HTMLElement> {
+  return basicContent("String");
+}
+
+function booleanContent(): JQuery<HTMLElement> {
+  return basicContent("Boolean");
+}
+
+function doubleContent(): JQuery<HTMLElement> {
+  return basicContent("Double");
+}
+
+function basicContent(contentType: string): JQuery<HTMLElement> {
+  return $("<tr/>", { id: contentType, class: "fieldValueContents" }).append(
+    textInputWithLabel("basicValue", "Value: ", "value input box")
+  );
+}
+
+function webImageContent(): JQuery<HTMLElement>[] {
+  return [
+    $("<tr/>", { id: "WebImage", class: "fieldValueContents" }).append(
+      textInputWithLabel("url", "Image url: ", "Image url input box")
+    ),
+    $("<tr/>").append(textInputWithLabel("altText", "[Recommended] Alt-text: ", "Alt-text input box"))
+  ];
+}
+
+function formattedNumberContent(): JQuery<HTMLElement>[] {
+  return [
+    $("<tr/>", { id: "FormattedNumber", class: "fieldValueContents" }).append(
+      textInputWithLabel("number", "Number: ", "Number input box")
+    ),
+    $("<tr/>").append(textInputWithLabel("format", "Format: ", "Format input box"))
+  ];
+}
+
+function unsupportedContent(): JQuery<HTMLElement> {
+  return $("<tr/>", { id: "Unsupported", class: "fieldValueContents" })
+    .append($("<td/>").append($("<label/>", { for: "unsupportedValue", class: "labels" }).text("unsupportedValue: ")))
+    .append(
+      $("<td/>").append(
+        $("<input>", {
+          class: "inputBox unsupportedValue",
+          disabled: true,
```

Comment:
Should this be false? I couldn't see any errors when entering a string as a Double type.